### PR TITLE
Adjust needle info in assert_screen_with_soft_timeout

### PR DESCRIPTION
**Pull request by jknphy**
Adjust needle info in assert_screen_with_soft_timeout

- Related ticket: https://progress.opensuse.org/issues/47387
- Needles: N/A
- Verification run: ~thinking how to test it...or where~
  - [sle-15-SP1-create_hdd_sled_gnome](http://rivera-workstation.suse.cz/tests/1824) calling it in this way: `assert_screen_with_soft_timeout([qw(inst-addon addon-products)], timeout => 0, soft_timeout => 0, bugref => 'bsc#1123963')`

### CODE REVIEW

File: lib/utils.pm [hunk 1]
```diff
@@ -534,9 +534,10 @@ Example:
 sub assert_screen_with_soft_timeout {
     my ($mustmatch, %args) = @_;
     # as in assert_screen
-    $args{timeout}             //= 30;
-    $args{soft_timeout}        //= 0;
-    $args{soft_failure_reason} //= $args{bugref} . ': needle(s) not found within ' . $args{soft_timeout};
+    $args{timeout}      //= 30;
+    $args{soft_timeout} //= 0;
+    my $needle_info = ref($mustmatch) eq "ARRAY" ? join(',', @$mustmatch) : $mustmatch;
+    $args{soft_failure_reason} //= $args{bugref} . ': needle(s) ' . $needle_info . ' not found within ' . $args{soft_timeout};
```
Comment (SergioAtSUSE) [inline lib/utils.pm]: I think here you can use " for readability

```suggestion
    $args{soft_failure_reason} //= $args{bugref} . ": needle(s) $needle_info not found within " . $args{soft_timeout};
```

Comment (jknphy) [reply lib/utils.pm]: I don't know if it is just my atom editor that `bugref` is not highlighted in that way in `$args{bugref}` but you're right is more compact...

Comment (rwx788) [inline lib/utils.pm]: Looks like code contained an error and it should be `$args{timeout}` here, instead of `$args{soft_failure_reason}`.

**Comment by rwx788** [comment]
As for your question, just trigger any test suite which executes this path and set timeout to 0, so it always soft-fails.

File: lib/utils.pm (hunk 1 above)
Comment (jknphy) [reply lib/utils.pm]: $args{soft_failure_reason} is used in the soft-failure, I think is ok? could you rephrase it, maybe I'm missing something.

Comment (rwx788) [reply lib/utils.pm]: You are right, I got confused as timeout should be bigger than soft_timeout. Sorry =)

**Comment by jknphy** [comment]
> As for your question, just trigger any test suite which executes this path and set timeout to 0, so it always soft-fails.
 I could not reproduce with timeouts 0, I have to check why...

**Comment by jknphy** [comment]
> As for your question, just trigger any test suite which executes this path and set timeout to 0, so it always soft-fails.

I could not reproduce with timeouts 0, I have to check why...

**Comment by okurz** [comment]
thank you!
